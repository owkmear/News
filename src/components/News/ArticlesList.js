import React from 'react';
import Article from './Article';

class ArticlesList extends React.Component {
  render() {
    const { data } = this.props;
    return (
      <div className="news__articles-list">
        {data.map(articleInfo =>
          <Article key={articleInfo.id} {...articleInfo} />
        )}
      </div>
    );
  }
}

export default ArticlesList;