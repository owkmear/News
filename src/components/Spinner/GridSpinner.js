import React from 'react';
import { GridLoader } from 'react-spinners';

class GridSpinner extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: true,
    };
  }
  render() {
    return <GridLoader color={'#337ab7'} loading={this.state.loading} />;
  }
}

export default GridSpinner;
