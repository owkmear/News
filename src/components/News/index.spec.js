import React from 'react';
import { shallow } from 'enzyme';
import { News } from './index';
import ArticlesList from './ArticlesList';
import Spinner from '../Spinner';

describe('News component', () => {
  const props = {
    data: [],
    isLoading: false,
    errorMessage: null,
    getNewsList: () => {},
  };

  describe('News component initial', () => {
    const mockgetNewsList = jest.fn();
    const nextProps = {
      ...props,
      getNewsList: mockgetNewsList,
    };
    const wrapper = shallow(<News {...nextProps} />);

    it('Render properly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('Dispatches the `getNewsList()` method is receives from props', () => {
      expect(mockgetNewsList).toHaveBeenCalled();
    });
  });

  describe('News component isLoading', () => {
    const nextProps = {
      ...props,
      isLoading: true,
    };
    const wrapper = shallow(<News {...nextProps} />);

    it('Render properly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('News component render <Spinner /> if isLoading is true', () => {
      expect(wrapper.find(Spinner)).toHaveLength(1);
    });
  });

  describe('News component render <ArticlesList />', () => {
    const nextProps = {
      ...props,
      data: [{ id: '2688074' }, { id: '2687983' }, { id: '2687985' }],
      isLoading: false,
    };

    const wrapper = shallow(<News {...nextProps} />);

    it('Render properly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('Should <ArticlesList /> exist', () => {
      expect(wrapper.find(ArticlesList)).toHaveLength(1);
    });
  });

  describe('News component with errorMessage', () => {
    const nextProps = {
      ...props,
      isLoading: false,
      errorMessage: 'Path does not exist',
    };

    const wrapper = shallow(<News {...nextProps} />);

    it('Render properly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('Render errorMessage', () => {
      expect(wrapper.find('.news__error').text()).toEqual(
        nextProps.errorMessage
      );
    });
  });
});
