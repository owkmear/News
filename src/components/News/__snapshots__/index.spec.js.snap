// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News component News component initial Render properly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-12 d-flex"
    >
      <div
        className="shadow-box shadow-box_round-big"
      >
        <div
          className="shadow-box__inner d-flex flex-column"
        >
          <section
            className="news"
          >
            <div
              className="row"
            >
              <div
                className="col"
              >
                <h2
                  className="block__header"
                >
                  News
                </h2>
                <p
                  className="news__empty"
                >
                  No news
                </p>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`News component News component isLoading Render properly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-12 d-flex"
    >
      <div
        className="shadow-box shadow-box_round-big"
      >
        <div
          className="shadow-box__inner d-flex flex-column"
        >
          <section
            className="news"
          >
            <div
              className="row"
            >
              <div
                className="col"
              >
                <h2
                  className="block__header"
                >
                  News
                </h2>
                <div
                  className="d-flex justify-content-center"
                >
                  <Spinner />
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`News component News component render <ArticlesList /> Render properly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-12 d-flex"
    >
      <div
        className="shadow-box shadow-box_round-big"
      >
        <div
          className="shadow-box__inner d-flex flex-column"
        >
          <section
            className="news"
          >
            <div
              className="row"
            >
              <div
                className="col"
              >
                <h2
                  className="block__header"
                >
                  News
                </h2>
                <ArticlesList
                  data={
                    Array [
                      Object {
                        "id": "2688074",
                      },
                      Object {
                        "id": "2687983",
                      },
                      Object {
                        "id": "2687985",
                      },
                    ]
                  }
                />
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`News component News component with errorMessage Render properly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-12 d-flex"
    >
      <div
        className="shadow-box shadow-box_round-big"
      >
        <div
          className="shadow-box__inner d-flex flex-column"
        >
          <section
            className="news"
          >
            <div
              className="row"
            >
              <div
                className="col"
              >
                <h2
                  className="block__header"
                >
                  News
                </h2>
                <div
                  className="news__error"
                >
                  Path does not exist
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
  </div>
</div>
`;
